### PR TITLE
Feat: Add custom variants

### PR DESCRIPTION
**Pull request by atopper**
- Add 'variants' text field
- move "Download" buttons lower (not really front and center while doing the mapping)
- move some things to be more globally accessible

image

### CODE REVIEW

File: modules/transformer.js [hunk 1]
```diff
@@ -52,6 +52,13 @@ export default class Transformer {
       const {
         type, variants, selectors, parse, insertMode = 'replace', params = {},
```
Comment (arumsey) [inline modules/transformer.js:53]: `variants` are already supported with each block rule. Why do we need a separate list of variants?

Comment (arumsey) [reply modules/transformer.js:53]: We can continue to discuss how the experience of actually applying variants during the mapping phase should look but I do not want to add extra complexity to the ruleset definition if it is not required.

Comment (arumsey) [reply modules/transformer.js:53]: Also, if we are trageting block collection blocks first we should confirm if any of those blocks even support variants before we invest too much time in this area early on.

File: modules/parsers/carousel.js [hunk 2]
```diff
@@ -33,6 +33,9 @@ function getCommonAncestor(elements) {
 export default function parse(el) {
   // a carousel will consist of one row for every image found
   const images = el.querySelectorAll('img');
+  if (images.length === 1 && images[0].children.length === 0) {
+    return [[images[0]]];
+  }
```
Comment (atopper) [inline modules/parsers/carousel.js:38]: In Hubble's 'article' page, there was a carousel with only 1 image assigned to it.  There were no commonParent's children.

**Review by arumsey — APPROVED** [review]

**Pull request merged**